### ISSUE TITLE
Key --blacklist fails

### ISSUE THREAD

**Issue by MikhailRokhin** [ValdikSS/GoodbyeDPI#452]
### Operating system / операционная система

Version 10.0.22631.3880

### Running as service / Запуск программы как сервис

I run it as a regular program / Запускаю программу обычным образом

### Describe the bug / Опишите ошибку программы

Even when --blacklist key is supplied, it affects connections to other addresses (looks like to all). 

For e.g., when googlevideo.com accesses should be processed only, it interferes with vk.com, Microsoft Store, Opera weather widget, etc. connections.

Actual for v0.2.2 .

Very heavy case:
goodbyedpi.exe --blacklist ..\googlevideo.com.txt -f 1 -e 1 --reverse-frag  

Partial work-around: remove Native fragmentation (splitting) & Fragments sending in reverse . 

To let other connections to other addresses succeed, one should stop the app, let the connections be established & start the app again.

### Additional information / Дополнительная информация

Thank you for development & support.

Joy, Happiness and Satisfaction! Resources for prosperous life!!)

**Comment by MikhailRokhin** [comment]
and it interferes with dns-over-https connections system-wide just for the command:

goodbyedpi.exe --blacklist ..\googlevideo.com.txt -f 1 -e 1

**Comment by ValdikSS** [comment]
Please provide traffic dump to the domains not listed in the blacklist

https://github.com/ValdikSS/GoodbyeDPI/wiki/How-to-make-traffic-dump-with-Wireshark

**Comment by MikhailRokhin** [comment]
This site can’t be reached
The webpage at https://www.youtube.com/live_chat?continuation=0ofMyAN-Gl5DaWtxSndvWVZVTk5Ra2t5YnpFdGVHOVdObWsxVGtzNGRWbE5jR05CRWd0WGQzcGtXbVV5Tmw5cVNSb1Q2cWpkdVFFTkNndFhkM3BrV21VeU5sOXFTU0FCTUFBJTNEMAGCAQYIBBgCIACIAQGgAd-7gL7M2YcDqAEAsgEA&authuser=0 might be temporarily down or it may have moved permanently to a new web address.

ERR_FAILED

**Comment by ValdikSS** [comment]
youtube.com is in blacklist.

**Comment by MikhailRokhin** [comment]
And there is persistent error for   0_russia_update_blacklist_file.cmd :

DISPLAY: 'blacklist' TYPE: DOWNLOAD STATE: TRANSIENT_ERROR
PRIORITY: NORMAL FILES: 0 / 1 BYTES: 0 / UNKNOWN
ERROR FILE:    https://antizapret.prostovpn.org/domains-export.txt -> C:\...\goodbyedpi\goodbyedpi-0.2.2\rus
ERROR CODE:    0x80072efe - Соединение с сервером было неожиданно прервано
ERROR CONTEXT: 0x00000005 - Ошибка при обработке удаленного файла.

**Comment by MikhailRokhin** [comment]
For ` goodbyedpi.exe -f 1 -e 1 --blacklist ..\googlevideo.com.txt --allow-no-sni -p --wrong-seq --wrong-chksum --dns-verb --dns-addr  8.8.8.8 ` : 
 
GoodbyeDPI v0.2.2: Passive DPI blocker and Active DPI circumvention utility
https://github.com/ValdikSS/GoodbyeDPI

WARNING: host  is less than 3 bytes, skipping
Loaded 1 hosts from file ..\googlevideo.com.txt
Block passive: 1
Fragment HTTP: 1
Fragment persistent HTTP: 0
Fragment HTTPS: 1
Native fragmentation (splitting): 0
Fragments sending in reverse: 0
hoSt: 0
Host no space: 0
Additional space: 0
Mix Host: 0
HTTP AllPorts: 0
HTTP Persistent Nowait: 0
DNS redirect: 1
DNSv6 redirect: 0
Allow missing SNI: 1
Fake requests, TTL: disabled (fixed: 0, auto: 0-0-0, min distance: 0)
Fake requests, wrong checksum: 1
Fake requests, wrong SEQ/ACK: 1
Max payload size: 0

Opening filter
Filter activated, GoodbyeDPI is now running!
[DNS] Error handling incoming packet: srcport = 53, dstport = 49830
[DNS] Error handling incoming packet: srcport = 53, dstport = 58133
[DNS] Error handling incoming packet: srcport = 53, dstport = 59981 ...

May that be result of ` Windows Defender Firewall ` set up to block all incoming connections? And on the network interface IPv4 is enabled only.

OS percepts ` goodbyedpi.exe ` as Windows Vista Operating system context, and Data execution prevention (DEP) is On.

Is there any known issue of ` WinDivert*.sys ` and/or ` goodbyedpi.exe ` with OS core isolation mode (any silent suppressions from ` Windows Defender ` vulnerable driver blocklist)?

**Comment by johnnyd-78** [comment]
> May that be result of `Windows Defender Firewall` set up to block all incoming connections? And on the network interface IPv4 is enabled only.

Проверил у себя, нет ничего не меняется, хоть с открытыми для списка програм, хоть с закрытыми для всех. Ничего, кроме IPv4 и нет, это ж Ростелеком )

> Is there any wildcard abilities for blacklists, like `*, ?, #, %` ... , grep expressions ( `*.googlevideo.com`, `*.?ooglevideo.com`) ?

googlevideo.com в блэклисте будет работать для googlevideo.com и для *.googlevideo.com А вообще WC никакие указывать возможности нет.
И странно думать, что прописывание одного этого домена вам починит ютуб, их там минимум штук 5 надо. Возможно, из-за того что у вас там всего 1 строка блэклист и не работает.

**Comment by johnnyd-78** [comment]
> And there is persistent error for `0_russia_update_blacklist_file.cmd` :

Все верно, домен заблокирован Ростелекомом, использование блэклиста russia-blacklist.txt в программе как раз эту проблему решает.

**Comment by MikhailRokhin** [comment]
> > And there is persistent error for `0_russia_update_blacklist_file.cmd` :
> 
> Все верно, домен заблокирован Ростелекомом, использование блэклиста russia-blacklist.txt в программе как раз эту проблему решает.

No, you are wrong about Rostelecom and domains: the only ` *.googlevideo.com ` is blocked and needs to be unblocked, and it works with regular VPN in selective mode fine. 

Rostelecom provider still allows access to web-interface of youtube.

**Comment by johnnyd-78** [comment]
> Rostelecom provider still allows access to web-interface of youtube.

По-русски можете общаться, вы русский, я русский, автор русский, что вы тут тренируетксть в английском, вам это какое-то удовольствие доставляет или что? )

Я не понимаю, что вы мне, клиенту Ростелекома рвссказываете, что он замедляет, а что нет, думаете я не вижу этого?

**Comment by johnnyd-78** [comment]
В общем, провел эксперимент - сделал txt с одним адресом googlevideo.com (даже перевод строки не добавлял) и запустил GDPI так:
-9 --blacklist ..\YTTest.txt
Ютуб конечно же не пашет нормально, превьюшек нет, аватарок нет, да и загрузка видео - вечная крутилка.
Открыл nexusmods.com - открывается, работает.
Сменил запуск на просто -9
Нексус начал выдавать Код ошибки: SSL_ERROR_RX_MALFORMED_SERVER_HELLO в Мозилле и ERR_SSL_PROTOCOL_ERROR в Хроме.

Вывод - блэклист работает, даже с 1 строкой и даже без перевода строки. Актуально для 0.2.2 и 0.2.3rc1 - проверил на обеих (на .2.2 естественно -9 нет, сменил на -6). Самое интересное - на 0.2.2 с -6 заработал ютуб в полном объеме, безо всяких блэклистов :D

**Comment by MikhailRokhin** [comment]
> В общем, провел эксперимент - сделал txt с одним адресом googlevideo.com (даже перевод строки не добавлял) и запустил GDPI так: -9 --blacklist ..\YTTest.txt Ютуб конечно же не пашет нормально, превьюшек нет, аватарок нет, да и загрузка видео - вечная крутилка. Открыл nexusmods.com - открывается, работает. Сменил запуск на просто -9 Нексус начал выдавать Код ошибки: SSL_ERROR_RX_MALFORMED_SERVER_HELLO в Мозилле и ERR_SSL_PROTOCOL_ERROR в Хроме.
> 
> Вывод - блэклист работает, даже с 1 строкой и даже без перевода строки. Актуально для 0.2.2 и 0.2.3rc1 - проверил на обеих (на .2.2 естественно -9 нет, сменил на -6). Самое интересное - на 0.2.2 с -6 заработал ютуб в полном объеме, безо всяких блэклистов :D

Do you know about different configurations of nodes within one provider (it was discussed in topics about Rostelecom either) ?

I just describe my case. In my case there is no need to unblock any domains, but ` googlevideo.com ` , 'cause access to web-interface of youtube is still available without any VPN/ADPI, directly. 

In my case, the best is ` -f 1 -e 1 ` fragmentation parameter.

For my case, I found meta-solution. Boot-strap connection to ` googlevideo.com ` with ` GDPI ` and then close ` GDPI ` to stop it interfering with browser's and system-wide Dns-Over-Https. In very lucky cases it allows to watch 1-2 hour videos without interruptions, but usually ` GDPI ` needs to be re-run & closed to reestablish connection to ` googlevideo.com ` after 30-60 minutes of watch.

I do more likely to blame TCP/IP stack, as I noticed better work of  ` GDPI ` after wake from hibernation, meaning it stops to interfere with other services & domains. But it happens from time to time, not on regular basis.

There is no change to good after update to recent Microsoft Windows [Version 10.0.22631.4037] .

**Comment by ValdikSS** [comment]
> meaning it stops to interfere with other services & domains. But it happens from time to time, not on regular basis.

Can't say for sure, but it could be that the issues you see are from the ISP (GoodbyeDPI is being detected on TSPU), not from the software itself.
